packages/components/src/components/Anchor/Anchor.tsx

```diff
@@ -1,12 +1,19 @@
 import React from "react";
 import { Text } from "../../primitives/Text";
 
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type AnchorAsOptions<Props = any> = React.ElementType<Props>;
+
 export interface AnchorProps
   extends Omit<React.AnchorHTMLAttributes<HTMLAnchorElement>, "color"> {
+  /** Sets the render element of the component. */
+  as?: AnchorAsOptions;
```

Comment:
Should we require this if `isExternal` is false? Feels easy for people to forget otherwise and to make sure people don't include it for external links?

Reply:
I'm not really in favor of this in the situation someone uses Pluto for something other than Next or React-Router. Maybe they just want a plain old link.

Reply:
Hmm any thoughts on how we can make sure people don't forget it then?

Reply:
We can add to the documentation. Or add tests on the app side when it happens.